storage/s3/mock.go

```diff
@@ -1,15 +1,45 @@
 package s3
 
 import (
+	"encoding/json"
+
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/stretchr/testify/mock"
 )
 
 // MockS3Client structure with s3API mock for testing
 type MockS3Client struct {
+	mock.Mock
 	s3iface.S3API
 }
 
 func (m *MockS3Client) PutObject(input *s3.PutObjectInput) (*s3.PutObjectOutput, error) {
-	return &s3.PutObjectOutput{}, nil
+	args := m.Called(input)
+
+	var obj s3.PutObjectOutput
+	objBytes, _ := json.Marshal(args.Get(0))
+	_ = json.Unmarshal(objBytes, &obj)
+
+	return &obj, args.Error(1)
+}
+
+func (m *MockS3Client) GetObject(input *s3.GetObjectInput) (*s3.GetObjectOutput, error) {
+	args := m.Called(input)
+
+	var obj s3.GetObjectOutput
+	objBytes, _ := json.Marshal(args.Get(0))
+	_ = json.Unmarshal(objBytes, &obj)
+
+	return &obj, args.Error(1)
+}
+
+func (m *MockS3Client) DeleteObject(input *s3.DeleteObjectInput) (*s3.DeleteObjectOutput, error) {
+	args := m.Called(input)
+
+	var obj s3.DeleteObjectOutput
+	objBytes, _ := json.Marshal(args.Get(0))
+	_ = json.Unmarshal(objBytes, &obj)
```

Comment:
```suggestion
	json.Unmarshal(objBytes, &obj)
```
If you don't handle the errors you can just omit this. The same applies to the upper functions.

Reply:
if a omit it the review fails

Reply:
You can add a comment line above the call, to ignore that check, like this:

```golang
//lint:ignore SA4006 we can safely ignore the error check here since it is a mock
json.Unmarshal(objbytes, &obj)
```

I'm not sure about the check code, here are all of the codes https://staticcheck.io/docs/checks use the appropriate one.

Reply:
@FilipNikolovski it is not the `statichcheck` that is not passing it is `kiseilk/errcheck` either revert the changes or maybe try to exclude mock file with [this](https://github.com/kisielk/errcheck#excluding-functions)